src/core/CorePermissions.sol

```diff
@@ -69,9 +69,13 @@ abstract contract CorePermissions is ICorePermissions, AccessControlEnumerableUp
         }
     }
 
-    function revokeRole(bytes32 role, address account) public override onlyRole(getRoleAdmin(role)) {
+    function revokeRole(bytes32 role, address account)
+        public
+        override(IAccessControlUpgradeable, AccessControlUpgradeable)
+        onlyRole(getRoleAdmin(role))
+    {
+        require(msg.sender != account, "NO_SELFREVOKE");
```

Comment:
nice, can you add a comment in here as to why we req this? so that there is always at least 1 governor

Reply:
actually - do we need limit this check to `GOVERN_ROLE`?
just realized it now applies to all the roles...

Reply:
ah right, because other cannot self-revoke anyway